### PR TITLE
Spring security

### PR DESCRIPTION
**Pull request by ChabVlad**

### CODE REVIEW

File: src/main/java/project/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -23,6 +24,7 @@
 @Tag(name = "Book controller", description = "endpoints for managing library")
 @RequiredArgsConstructor
 @RestController
+//@PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
```
Comment (ahoienko) [inline src/main/java/project/bookstore/controller/BookController.java]: Please don`t push commented code on remote

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -2,8 +2,10 @@ spring.application.name=BookStore
 spring.datasource.url=jdbc:mysql://localhost/book_store?serverTimezone=UTC
 spring.datasource.driverClassName=com.mysql.cj.jdbc.Driver
 spring.datasource.username=root
-spring.datasource.password=pass
+spring.datasource.password=!Vladius080197
```
Comment (ahoienko) [inline src/main/resources/application.properties]: It is unsafe to push passwords on git, you should left this field blank or use env variable

File: src/main/java/project/bookstore/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -15,13 +20,19 @@
 public class UserServiceImpl implements UserService {
     private final UserMapper userMapper;
     private final UserRepo userRepository;
+    private final PasswordEncoder encoder;
+    private final RoleRepo roleRepo;
 
     @Override
     public UserResponseDto register(UserRegistrationRequestDto requestDto) {
-        if (userRepository.existsUserByEmail(requestDto.getEmail())) {
+        Role role = roleRepo.findByRole(RoleName.ROLE_ADMIN);
```
Comment (ahoienko) [inline src/main/java/project/bookstore/service/impl/UserServiceImpl.java]: Why any user that registers is admin?

**Review by ahoienko — CHANGES_REQUESTED** [review]
Good job! left minor comments

File: src/main/java/project/bookstore/model/User.java [hunk 4]
```diff
@@ -5,30 +5,52 @@
...
 import lombok.Setter;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
 
 @Entity
 @Table(name = "users")
 @Getter
 @Setter
-public class User {
+public class User implements UserDetails {
```
Comment (liliia-ponomarenko) [inline src/main/java/project/bookstore/model/User.java:24]: It is better to override all methods with UserDetails. In particular, the isEnabled method will have its own implementation in your solution ;)

File: src/main/java/project/bookstore/repository/role/RoleRepo.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package project.bookstore.repository.role;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import project.bookstore.model.Role;
+import project.bookstore.model.RoleName;
+
+public interface RoleRepo extends JpaRepository<Role, Integer> {
```
Comment (liliia-ponomarenko) [inline src/main/java/project/bookstore/repository/role/RoleRepo.java]:
```suggestion
public interface RoleRepository extends JpaRepository<Role, Long> {
```
Don't use abbreviations in the names of variables, classes, and methods.
Also, you used Long as an id variable type.

File: src/main/java/project/bookstore/repository/user/UserRepo.java [hunk 6]
```diff
@@ -1,10 +1,13 @@
...
+import java.util.Optional;
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.stereotype.Repository;
 import project.bookstore.model.User;
 
-@Repository
-public interface UserRepo extends JpaRepository<User, Integer> {
-    Boolean existsUserByEmail(String email);
+public interface UserRepo extends JpaRepository<User, Long> {
```
Comment (liliia-ponomarenko) [inline src/main/java/project/bookstore/repository/user/UserRepo.java]:
```suggestion
public interface UserRepository extends JpaRepository<User, Long> {
```

File: src/main/resources/db/changelog/changes/06-add-isDeleted-to-users.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+      changes:
+        - addColumn:
+            tableName: users
+            columns:
+              - column:
+                  name: is_deleted
+                  type: tinyint
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/changes/06-add-isDeleted-to-users.yaml]: Left one empty line at the end of the file  ;)

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution ;)

File: src/main/java/project/bookstore/config/SecurityConfig.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Bean
+    public SecurityFilterChain getSecurityFilterChain(HttpSecurity http) throws Exception {
+        return http
+                .cors(AbstractHttpConfigurer::disable)
+                .csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(
+                        auth -> auth
+                                .requestMatchers(
+                                        HttpMethod.POST,
```
Comment (Elena-Bruyako) [inline src/main/java/project/bookstore/config/SecurityConfig.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
one comment

**Review by liliia-ponomarenko — APPROVED** [review]
Awesome 😎

**Pull request merged**